dbgd.c

```diff
@@ -284,3 +286,81 @@ static int dbgd_show_front_screen(Dbg__Request *req, Dbg__Response *res)
 
     return DBG__RESPONSE__TYPE__OK;
 }
+
+static __attribute__((__stdcall__)) uint32_t test(uint32_t a, uint32_t b) {
+  debugPrint("test(%d, %d)\n", a, b);
+  return a + b;
+}
+
+static int dbgd_call(Dbg__Request *req, Dbg__Response *res)
+{
+    if (!req->has_address)
+        return DBG__RESPONSE__TYPE__ERROR_INCOMPLETE_REQUEST;
+
+    size_t stack_size = 0x1000;
+    uint8_t* stack_data = (uint8_t*)malloc(stack_size); // FIXME: Calculate correct size
+
+    static uint32_t stack_pointer;
+    static uint32_t stack_backup;
+    static uint32_t address;
+ 
+    stack_pointer = (uint32_t)&stack_data[stack_size];
+    address = req->address;
+
+    // Push stack contents
+
+    if (req->has_data) {
+      stack_pointer -= req->data.len;
+      memcpy((void*)stack_pointer, req->data.data, req->data.len);
+    }
+
+    // This is where the call will happen
+
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // EAX
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // ECX
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // EDX
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // EBX
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // ESP (Will be ignored)
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // EBP
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // ESI
+    stack_pointer -= 4;
+    *(uint32_t*)(stack_pointer) = 0; // EDI
```

Comment:
Remove this. We don't have to set regs. If user needs this, they can just add it on their own.